pkg/endpoint/local_endpoint.go

```
@@ -171,42 +174,67 @@ func GetLocalSpec(ctx context.Context, submSpec *types.SubmarinerSpecification,
 	}
 
 	endpointSpec := &submv1.EndpointSpec{
-		CableName:     fmt.Sprintf("submariner-cable-%s-%s", submSpec.ClusterID, strings.ReplaceAll(privateIP, ".", "-")),
 		ClusterID:     submSpec.ClusterID,
 		Hostname:      hostname,
-		PrivateIP:     privateIP,
 		NATEnabled:    submSpec.NATEnabled,
 		Subnets:       localSubnets,
 		Backend:       submSpec.CableDriver,
 		BackendConfig: backendConfig,
 	}
 
-	publicIP, resolver, err := getPublicIP(submSpec, k8sClient, backendConfig, airGappedDeployment)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not determine public IP")
+	for _, privateIP := range privateIPs {
+		endpointSpec.SetPrivateIP(privateIP)
 	}
 
-	logger.Infof("Obtained local endpoint public IP %q using resolver %q", publicIP, resolver)
+	endpointSpec.CableName = fmt.Sprintf("submariner-cable-%s-%s", submSpec.ClusterID,
+		strings.ReplaceAll(endpointSpec.GetPrivateIP(k8snet.IPv4), ".", "-"))
+	publicIPs, err := getPublicIPs(submSpec, k8sClient, backendConfig, airGappedDeployment)
+
+	if len(publicIPs) == 0 {
+		return nil, errors.Wrap(err, "could not determine public IP")
+	}
```

Again this silences the error if there's only one family that had a failure. It should fail if any error occurs. Also we don't need `getPublicIPs` - use `getPublicIP` and pass in the family as should the watcher:

```
for _, family := range submSpec.GetIPFamilies() {
    publicIP, resolver, err := getPublicIP(family, submSpec, k8sClient, backendConfig, airGappedDeployment)
    if err != nil {
        return nil, errors.Wrapf(err, "could not determine public IP for IP family %v", family)
    }

    logger.Infof("Obtained local endpoint public IP %q using resolver %q for family %v", publicIP, resolver, family)

    endpointSpec.SetPublicIP(publicIP)
}

```

---

> Again this silences the error if there's only one family that had a failure. It should fail if any error occurs

Hmm, not 100% sure about that, 
So, in case one family failed in a dual-stack case, you think we should fail the gateway and not continue with the one that succeeded, right?

---

Not entirely sure but currently that's the behavior with just IP4 and would be the behavior with just IPv6. What would/should be the overall behavior in the GW if some aspect(s) of IPv6 failed in dual-stack? Would some features be available but not others? Or do we somehow disable IPv6 support across the board?